### PR TITLE
Add queue priority design doc

### PR DESCRIPTION
**Pull request by TaiPark**
Doc for issue: https://github.com/volcano-sh/volcano/issues/3542

### CODE REVIEW

File: docs/design/queue-priority.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+The current queue prioritization in Volcano is based on the principle that a lower share value indicates a higher priority for resource allocation during scheduling.
+
+However, based on user feedback and practical scenarios, there is a strong preference for a more straightforward approach to establishing queue priorities. Explicitly assigning priorities provides users with clear and immediate control over the order in which their queues are serviced.
+
+While Volcano's current method of using queue shares effectively determines priority, there is a clear demand for a more intuitive and user-friendly approach to setting and managing queue order. Enabling users to directly assign and adjust queue priorities in the specifications would simplify queue management significantly.
+
+## Implementation
+
+### Data Structure
```
Comment (Monokaix) [inline docs/design/queue-priority.md:15]: API Change is better: ）

Comment (TaiPark) [reply docs/design/queue-priority.md]: DONE

File: docs/design/queue-priority.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Assuming the overused queues do not enter the scheduling process, the `QueueOrderFn` in the capacity and proportion plugins will follow this sorting logic:
+- Prioritize by priority: queues with higher priorities will be positioned at the front of the PriorityQueue and processed first.
+- If the priority values are identical, the current method of comparing queue shares will be applied.
+
+### Reclaim
+
+The `reclaim` action is implemented for the reclaim logic between queues. Queue prioirty should be considered when reclaiming across queues.
+- The queue reclaiming should be orderly processed from the highest priority queue to the lowest, as guaranteed by `QueueOrderFn`.
+- For the victim PriorityQueue construction, considering two victims with different job id, the one with the lower queue priority should be selected as the victim.
```
Comment (Monokaix) [inline docs/design/queue-priority.md:37]: I think we should also clarify that if no enough resoures are reclaimed, higher queues' jobs will also be reclaimed.

Comment (TaiPark) [reply docs/design/queue-priority.md]: DONE

**Comment by Monokaix** [comment]
Please squash to one commit.

File: docs/design/queue-priority.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Implementation
+
+### API Change
+
+Add a `priority` attribute to the spec of `queues.scheduling.volcano.sh`. The priority attribute controls the order of queues in `capacity` and `proportion` plugins that implement QueueOrderFn.
+```
+spec:
+    ...
+    priority:
```
Comment (hwdef) [inline docs/design/queue-priority.md:21]: Why not use k8s native priorityClassName

**Pull request merged**